lib/src/serializers/json_api.dart

```diff
@@ -249,7 +272,17 @@ class JsonApiDocument {
           _isAttributeError(error, attributeName) && _hasErrorDetail(error))
       : false;
 
-  List<String> errorsFor(String attributeName) => List<String>.from(errors
+  List<String> errorsFor(String attributeName) {
+    var key = 'errorsFor:${attributeName}';
+    List<String>? ids = _idsCache[key];
+    if (ids == null) {
+      ids = _errorsFor(attributeName);
```

Comment:
@halil81 The `_errorsFor` method returns a `List` of error messages, not IDs. I agree to reuse the same cache, however, let's rename it from `_idsCache` to `_stringsCache`.

Reply:
@algodave I fixed variable names. Thanks!